### PR TITLE
Feature/62 UI wiremock integration

### PR DESCRIPTION
**Pull request by matt-wills**
### Consent details display using local docker with wiremock

This PR covers the necessary changes to get the consent details to display in the UI, when running against docker and wiremock locally.

I have temporarily made the `ssoToken` optional in the call to retrieve consent details. I need to investigate this further, but I've added a TODO for now.

I now need to turn my attention to the consent decision endpoint.

### CODE REVIEW

File: securebanking-openbanking-uk-rcs-sample/docker/config/securebanking-openbanking-uk-rcs-docker.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+### Spring Configuration for securebanking-openbanking-uk-rcs running in Docker
+
+dns:
+  hosts:
+    root: localhost
```
Comment (matt-wills) [inline securebanking-openbanking-uk-rcs-sample/docker/config/securebanking-openbanking-uk-rcs-docker.yml:21]: We need to have a discussion about this, but I've followed the same pattern as OB toolkit for now.

File: securebanking-openbanking-uk-rcs-sample/docker/docker-compose-git.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    depends_on:
+      - securebanking-config-server
+      - wiremock
+      - rs-simulator
+    links:
+      - securebanking-config-server
+      - wiremock
+      - rs-simulator
+
+  rs-simulator:
```
Comment (matt-wills) [inline securebanking-openbanking-uk-rcs-sample/docker/docker-compose-git.yml:48]: This uses the "real" RS simulator, with wiremock stubbing IDM, AM and JWKMS.

File: securebanking-openbanking-uk-rcs-sample/docker/config/securebanking-openbanking-uk-rcs-docker.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+rcs:
+  issuerId: forgerock-rcs
+
+rs:
+  internal-port: 8080
+  base-url: http://rs-simulator:${rs.internal-port}
+
+idm:
+  internal-port: 9080
+  base-url: http://wiremock:${idm.internal-port}/idm
```
Comment (matt-wills) [inline securebanking-openbanking-uk-rcs-sample/docker/config/securebanking-openbanking-uk-rcs-docker.yml:36]: IDM, JWKMS and AM represented by wiremock.

File: securebanking-openbanking-uk-rcs-sample/docker/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
+#
```
Comment (matt-wills) [inline securebanking-openbanking-uk-rcs-sample/docker/docker-compose.yml:1]: The default docker-compose file, which retrieves config using the native file system based profile.

File: securebanking-openbanking-uk-rcs-sample/docker/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    depends_on:
+      - securebanking-config-server
+      - wiremock
+      - rs-simulator
+    links:
+      - securebanking-config-server
+      - wiremock
+      - rs-simulator
+
+  rs-simulator:
```
Comment (matt-wills) [inline securebanking-openbanking-uk-rcs-sample/docker/docker-compose.yml:50]: The "real" RS simulator which uses the `securebanking-openbanking-uk-rs-docker.yml` config file above.

File: securebanking-openbanking-uk-rcs-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/web/CorsFilter.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                log.warn("Origin header host [{}] does not match the expected host root [{}]", originUri.getHost(), hostRoot);
+                response.setStatus(SC_UNAUTHORIZED);
+                return;
+            }
+
+            response.addHeader("Access-Control-Allow-Origin", originHeader);
+            response.addHeader("Access-Control-Allow-Methods", ALLOWED_METHODS);
+            response.addHeader("Access-Control-Max-Age", MAX_AGE);
+            response.addHeader("Access-Control-Allow-Headers", ALLOWED_HEADERS);
+            response.addHeader("Access-Control-Allow-Credentials", ALLOWED_CREDENTIALS.toString());
```
Comment (matt-wills) [inline securebanking-openbanking-uk-rcs-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/web/CorsFilter.java:75]: Required response headers for CORS.

**Review by jorgesanchezperez — APPROVED** [review]

**Pull request merged**